src/gui/widgets/Knob.cpp

```diff
@@ -450,31 +501,38 @@ void Knob::drawKnob( QPainter * _p )
 	_p->drawImage( 0, 0, m_cache );
 }
 
-void Knob::paintEvent( QPaintEvent * _me )
+void Knob::drawLabel(QPainter& p)
 {
-	QPainter p( this );
-
-	drawKnob( &p );
 	if( !m_label.isEmpty() )
 	{
 		if (!m_isHtmlLabel)
 		{
-			p.setFont(adjustedToPixelSize(p.font(), SMALL_FONT_SIZE));
+			auto fm = p.fontMetrics();
+			const auto x = (width() - horizontalAdvance(fm, m_label)) / 2;
+			const auto descent = legacyMode() ? 2 : fm.descent();
+			const auto y = height() - descent; 
+
 			p.setPen(textColor());
-			p.drawText(width() / 2 -
-				horizontalAdvance(p.fontMetrics(), m_label) / 2,
-				height() - 2, m_label);
+			p.drawText(x, y, m_label);
 		}
 		else
 		{
 			// TODO setHtmlLabel is never called so this will never be executed. Remove functionality?
-			m_tdRenderer->setDefaultFont(adjustedToPixelSize(p.font(), SMALL_FONT_SIZE));
+			m_tdRenderer->setDefaultFont(font());
```

Comment:
Is this even needed?

Reply:
As mentioned in the (previously existing) TODO in line 520 the method `Knob::setHtmlLabel` is never called anywhere so it might as well get removed. However, I thought that it would not really be something for this PR.

Reply:
Does it still make sense to remove it here?

Reply:
The entire html label thing if it's not used.